### PR TITLE
[MINOR][SQL] Remove unimplemented instances in XML Data Source

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR removes unimplemented instances in XML Data Source. They are presumably copied from JSON/CSV Data Source but they are not implemented yet. 

### Why are the changes needed?

They are unreachable code.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Existing CI in this PR should test them out.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @sandeep-katta @shujingyang-db

**Review by beliefer — APPROVED** [review]
LGTM.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM (Pending CIs)

**Review by dongjoon-hyun — COMMENTED** [review]
Could you fix `unused import` errors?
```
[error] /home/runner/work/spark/spark/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/xml/StaxXmlParser.scala:43:37: Unused import
[error] Applicable -Wconf / @nowarn filters for this fatal warning: msg=<part of the message>, cat=unused-imports, site=org.apache.spark.sql.catalyst.xml
[error] import org.apache.spark.sql.sources.Filter
[error]                                     ^
[error] /home/runner/work/spark/spark/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/xml/XmlInferSchema.scala:34:50: Unused import
[error] Applicable -Wconf / @nowarn filters for this fatal warning: msg=<part of the message>, cat=unused-imports, site=org.apache.spark.sql.catalyst.xml
[error] import org.apache.spark.sql.catalyst.expressions.ExprUtils
[error]                               
```

**Comment by HyukjinKwon** [comment]
Will do!

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/xml/StaxXmlParser.scala [hunk 1]
```diff
@@ -40,16 +40,12 @@ import org.apache.spark.sql.catalyst.util.LegacyDateFormats.FAST_DATE_FORMAT
...
 import org.apache.spark.sql.errors.QueryExecutionErrors
 import org.apache.spark.sql.internal.SQLConf
-import org.apache.spark.sql.sources.Filter
 import org.apache.spark.sql.types._
 import org.apache.spark.unsafe.types.UTF8String
 
 class StaxXmlParser(
     schema: StructType,
-    val options: XmlOptions,
-    filters: Seq[Filter] = Seq.empty) extends Logging {
```
Comment (dongjoon-hyun) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/xml/StaxXmlParser.scala:50]: This constructor seems to be used . Shall we recover this?
```
[error] /home/runner/work/spark/spark/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/xml/XmlFileFormat.scala:127:9: too many arguments (found 3, expected 2) for constructor StaxXmlParser: (schema: org.apache.spark.sql.types.StructType, options: org.apache.spark.sql.catalyst.xml.XmlOptions): org.apache.spark.sql.catalyst.xml.StaxXmlParser
[error]         actualFilters)
[error]         ^
```

**Comment by sandeep-katta** [comment]
LGTM

**Review by dongjoon-hyun — APPROVED** [review]

**Review by LuciferYang — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**